google/cloud/pubsub/publisher_connection.cc

```diff
@@ -138,13 +138,13 @@ std::shared_ptr<pubsub::PublisherConnection> MakePublisherConnection(
     if (options.message_ordering()) {
       auto factory = [topic, options, sink, cq](std::string const& key) {
         return BatchingPublisherConnection::Create(
-            topic, options, key, SequentialBatchSink::Create(std::move(sink)),
-            cq);
+            std::move(topic), std::move(options), key,
```

Comment:
I do not think you want to `std::move` here. `factory` may be called multiple times, once for each ordering key that is discovered (dynamically).  You want a copy, and note that they are captured by copy too.

Reply:
Fixed.